### PR TITLE
Fixed empty.yaml file, fixed gts

### PR DESCRIPTION
**Pull request by serraramiro1**
Corrected origin, fixed gts to refer to the global map frame
image

### CODE REVIEW

File: ca_gazebo/worlds/empty.world [hunk 1]
```diff
@@ -1,42 +1,13 @@
-<?xml version="1.0"?>
-<sdf version="1.6">
+<?xml version="1.0" ?>
+<sdf version="1.5">
```
Comment (eborghi10) [inline ca_gazebo/worlds/empty.world]: Why did you change this line? I don't think it's really necessary?

File: ca_gazebo/worlds/empty.world [hunk 2]
```diff
@@ -1,42 +1,13 @@
...
     <!-- A global light source -->
     <include>
       <uri>model://sun</uri>
     </include>
-
     <!-- A ground plane -->
     <include>
       <uri>model://ground_plane</uri>
     </include>
-
```
Comment (eborghi10) [inline ca_gazebo/worlds/empty.world:13]: Why did you delete this? Are the default values?

**Review by eborghi10 — CHANGES_REQUESTED** [review]
Thanks for pushing this! I made two comments.

File: ca_gazebo/worlds/empty.world (hunk 2 above)
Comment (serraramiro1) [reply ca_gazebo/worlds/empty.world:13]: I took another `.world` file from another project, everything is working like it should now.

File: ca_gazebo/worlds/empty.world (hunk 1 above)
Comment (serraramiro1) [reply ca_gazebo/worlds/empty.world]: Reverted.

**Comment by serraramiro1** [comment]
## Working like a charm, please merge. @eborghi10 @ernestmc

File: ca_gazebo/launch/create_empty_world.launch [hunk 3]
```diff
@@ -4,6 +4,13 @@
   	<arg name="paused" value="$(optenv PAUSED false)"/>
 	<arg name="gui" value="$(optenv GUI true)"/>
 	<arg name="debug" value="$(optenv DEBUG false)"/>
+	<param name="create1/amcl/initial_pose_x" value="-3"/>
```
Comment (eborghi10) [inline ca_gazebo/launch/create_empty_world.launch]: There is one issue here:

When the `env` is not `empty`, these parameters will override the ones from the other launch files. For that, we should include them only if env == empty. A way of doing this is including these parameters in another launch file. Do you have a better idea?

**Review by eborghi10 — CHANGES_REQUESTED** [review]

File: ca_gazebo/launch/create_empty_world.launch (hunk 3 above)
Comment (serraramiro1) [reply ca_gazebo/launch/create_empty_world.launch]: I guess i found a workaround, maybe you can check it? It's working now

Comment (eborghi10) [reply ca_gazebo/launch/create_empty_world.launch]: Seems to be better than my approach!

**Comment by eborghi10** [comment]
Regarding the approach of disabling both `SLAM` and `AMCL`, consider deleting those environment variables and create a new one called `NAVIGATION` that can receive strings ("amcl", "slam"), and if it's not set, don't call the navigation launch file at all.

**Comment by serraramiro1** [comment]
## Adressed, please merge @eborghi10

File: navigation/ca_move_base/launch/navigate.launch [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <launch>
   <arg name="robot_name"/>
-  <arg name="slam" default="$(optenv SLAM false)"/>
+  <arg name="navigation" default="$(optenv NAVIGATION amcl)"/>
```
Comment (ernestmc) [inline navigation/ca_move_base/launch/navigate.launch]: amcl is a localization algorithm, not a navigation. I would recommend to change the name of the variable.

File: navigation/ca_move_base/launch/navigate.launch [hunk 5]
```diff
@@ -11,12 +11,12 @@
...
 
 	<!-- AMCL -->
-  <include if="$(eval arg('slam') == False)" file="$(find ca_move_base)/launch/amcl.launch">
+  <include if="$(eval str(arg('navigation')) == 'amcl')" file="$(find ca_move_base)/launch/amcl.launch">
     <arg name="ns" value="$(arg robot_name)"/>
   </include>
 
   <!-- slam_gmapping -->
-  <include if="$(eval arg('slam') == True)" file="$(find ca_slam)/launch/slam_gmapping.launch">
+  <include if="$(eval str(arg('navigation')) == 'slam')" file="$(find ca_slam)/launch/slam_gmapping.launch">
```
Comment (ernestmc) [inline navigation/ca_move_base/launch/navigate.launch]: Please add a case when the variable is `none` (neither amcl or slam) so no localization is loaded, in case we don't need it.

**Review by ernestmc — CHANGES_REQUESTED** [review]
Left some comments.

File: navigation/ca_move_base/launch/navigate.launch (hunk 4 above)
Comment (eborghi10) [reply navigation/ca_move_base/launch/navigate.launch]: I know but I don't have an idea now. Do you?

Comment (ernestmc) [reply navigation/ca_move_base/launch/navigate.launch]: maybe: "localization_package=", "use_localization=" ?

**Comment by serraramiro1** [comment]
Changed the NAVIGATION env to `LOCALIZATION`; as of right now the approach does the following:
*If nothing is specified, `navigate.launch` isn't called at all, what means no `AMCL` nor `SLAM`
*If LOCALIZATION=slam is set, `SLAM` is called
*If LOCALIZATION=amcl is set, `AMCL` is called

Is this the intended behaviour? @eborghi10 @ernestmc

**Comment by serraramiro1** [comment]
Please review @eborghi10

**Review by eborghi10 — APPROVED** [review]
Tested and working great. Thanks for pushing this! I'll update the documentation.

One thing to take into account is that `map_server` should only run when `LOCALIZATION=amcl`.
